### PR TITLE
Lps 182293 | Add automation test for LPS-177798 operate a custom object entry and it's related system objects in a single request

### PR DESCRIPTION
**Pull request by yanan**
Background:
- Add automation tests which without integration tests for Story LPS-177798

Jira Ticket
- https://issues.liferay.com/browse/LPS-182293

Tips:
- I've rebased to [LPS-180845](https://github.com/brianchandotcom/liferay-portal/pull/133758) which is not merged yet

### CODE REVIEW

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
The 3 failed tests are about Patch which we already have a related ticket: LPS-182396

**Review by Hijky — APPROVED** [review]

**Review by magjed4289 — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro [hunk 1]
```diff
@@ -97,45 +104,37 @@ definition {
...
-
-				if (isSet(relatedEntryExternalReferenceCode)) {
-					var relatedObjectEntry = StringUtil.add("${relatedObjectEntry},", "\"externalReferenceCode\": \"${relatedEntryExternalReferenceCode}${i}\"", "");
-				}
-
-				var j = ${i} + 1;
-
-				if (${j} == ${size}) {
-					var entryBody = StringUtil.add(${entryBody}, "{${relatedObjectEntry}}]", "");
+			if (${relatedSystemUser} == "true") {
```
[the remark this answers is not in the document]
Comment (yanan) [reply modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro:107]: good idea, I'll update and reslove the conflicts!

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
I've checked the only failed test, and created a bug ticket here [LPS-182826](https://issues.liferay.com/browse/LPS-182826) 
- OperateCustomObjectAndRelatedSystemObjectsWithinSingleRequest#CanUpdateCustomAndRelatedSystemObjectInManyToOneRelationshipByPatch
@magjed4289 would you like me to disable the failed tests until LPS-182826 fixed? And please help check if any updates need before forward? Thanks :heart:

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
ci:forward

**Pull request closed without merging**